src/main/java/com/minecolonies/coremod/placementhandlers/PlacementHandlers.java

```diff
@@ -449,11 +456,16 @@ public Object handle(
                 placer.handleBuildingOverBlock(pos);
             }
 
-            if (!world.setBlockState(pos, blockState, 0x03))
+            if (!world.setBlockState(pos, blockState, UPDATE_FLAG))
             {
                 return ActionProcessingResult.DENY;
             }
 
+            if(placer != null)
+            {
+                placer.handleTileEntityPlacement(pos);
+            }
+
             return blockState;
         }
     }
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce this class from 32 to the maximum allowed 30 or externalize it in a public class. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)